metaseq/dataclass/configs.py

```diff
@@ -737,3 +749,56 @@ class MetaseqConfig(MetaseqDataclass):
     lr_scheduler: Any = MISSING
     bpe: Any = MISSING
     tokenizer: Any = None
+
+
+class DynamicConfig:
+    """
+    can be used instead of redis to store updatable key-value
+
+    test:
+        data = {"a": 1}
+        with open('data.json', 'w') as fp:
+            json.dump(data, fp)
+        dcfg = DynoCfg(json_file_path = "data.json", timeout = 1)
+        print(dcfg["a"]) # > 1
+        data = {"a": 4}
+        with open('data.json', 'w') as fp:
+            json.dump(data, fp)
+        print(dcfg["a"]) # > 1
+        time.sleep(1)
+        print(dcfg["a"]) # > 4
+    """
+
+    default_state = {"force_profile": False}
+
+    def __init__(self, json_file_path=None, timeout=30):  # timeout in sec
+        self.data = DynamicConfig.default_state
+        self.timeout = timeout
+        self.timer_start = 0
+        self.json_file_path = json_file_path
+        self.refresh()
+
+    def refresh(self):
+        if self.json_file_path is not None:
+            # if data have not been updated in a while, reload from the file
+            # can speed up if were to calculate hash sum of the file to monitor updates
+            timer_value = time.time() - self.timer_start
+            if timer_value > self.timeout:
+                try:
+                    with open(self.json_file_path, "r") as json_file:
+                        self.data = json.load(json_file)
+                    self.timer_start = time.time()
+                except json.JSONDecodeError as jsonerror:
+                    logging.warning(
+                        f"""Error refreshing dynamic config: reading file {json_file}
+                    resulted in JSONDecodeError {jsonerror}"""
+                    )
+                except IOError as ioerror:
+                    logging.warning(
+                        f"""Error refreshing dynamic config: reading file {json_file}
+                    resulted in IOError {ioerror}"""
+                    )
```

Comment:
Could you add 3 tests for this?

1. Test that a malformed (truncated) file catches the error.
2. Test that an empty file catches the error.
3. Test that a file that doesn't exist catches the error.

My concern is that, with my limited knowledge of python exception types, a different exception type could be generated in a rare situation and impact the big training at an unexpected/inopportune time.